cluster-app-values-schema/README.md

```diff
@@ -0,0 +1,283 @@
+- The title MUST NOT contain punctuation marks, leading or trailing white space, control characters, tabs, nor multiple whitespaces in a row.
+- If the title annotates a property that is part of another object, the title SHOULD NOT include the parent property name, to avoid repetition.
+  - Example: with an object `/controlPlane` that is titled `Control plane`, the property  `/controlPlane/availabilityZones` should be titled `Availability zones`, not `Control plane availability zones`.
+
+### R5: Properties should have descriptions {#r5}
+
+Each property SHOULD be annotated and via the `description` keyword.
+
+Example:
+
+```json
+{
+  ...
+  "properties": {
+    "name": {
+        "type": "string",
+        "description": "Identifies this cluster uniquely within the installation.",
+        ...
+    }
+  }
+}
+```
+
+If a description is given, additional requirements apply to the value:
+
+- Description content MUST NOT include any line breaks.
+- Descriptions MUST NOT contain formatting code like e.g. Markdown or HTML, leading or trailing white space, control characters, tabs, nor multiple whitespaces in a row.
+- Descriptions MUST use sentence case capitalization and punctuation.
+- The description SHOULD NOT repeat the property name or title. Correct: "Identifies this cluster uniquely within the installation.", incorrect: "Cluster name identifies this cluster uniquely within the installation.".
+- Descriptions SHOULD be between 50 and 200 characters long.
+- Descriptions SHOULD be written in simple language.
+
+### R6: Properties should provide examples {#r6}
+
+Each property SHOULD provide one or more examples using the `examples` keyword.
+
+Example:
+
+```json
+{
+  ...
+  "properties": {
+    "name": {
+        "type": "string",
+        "examples": ["devel001"],
+        ...
+    }
+  }
+}
+```
+
+Best practices:
+
+- Ideally, provide one example.
+- Provide more than one example if you want to highlight that very different values are accepted.
+- Provide no more than five examples
+
+TODO
+
+- We could decide to use the examples for testing purposes, replacing ci-values.yaml and the likes. In that case, we should make it a MUST requirement.
+
+### R7: Constrain values as much as possible {#r7}
+
+Property schema SHOULD explicitly restrict values as much as possible.
+
+There are several ways this requirement can/has to be fulfilled, depending on the property type and the use case.
+
+Properties of type `string`:
+
+- Use of the keywords `minLength` and `maxLength` to specify the valid length.
+- If applicable, use the `pattern` keyword to specify a regular expression for validation. This is useful for example to restrict the value to ASCII characters, or to lowercase etc.
+- Use the `format` keyword to restrict the property to a common string format, like a URL or an email address. Watch out to use only formats which are available in the JSON schema dialect used (see also R1).
+- Use `enum` if only certain known values can be valid.
```

Comment:
Can we make our requirements here more specific?
E.g.
Properties of type `string`...

- ...SHOULD use `minLength` and `maxLength` to specify the valid length, unless `format` or `enum` is already used.
-  ...SHOULD have one of the following properties: `pattern`, `format` or `enum`.

Reply:
Clarification coming with the next commit.